pkg/slayers/path/scion/info_hop_setter_lemmas.gobra

```diff
@@ -0,0 +1,519 @@
+// Copyright 2022 ETH Zurich
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +gobra
+
+package scion
+
+import (
+	"github.com/scionproto/scion/pkg/slayers/path"
+	. "verification/utils/definitions"
+	sl "verification/utils/slices"
+	"verification/io"
+)
+
+ghost
+decreases
+pure func AbsSetCurrSeg(seg io.IO_seg3, info path.IntermediateAbsInfoField) (io.IO_seg3) {
+	return io.IO_seg3_ {
+		info.AInfo,
+		info.UInfo,
+		info.ConsDir,
+		info.Peer,
+		seg.Past,
+		seg.Future,
+		seg.History,
+	}
+}
```

Comment:
This function feels out of place. I would rather have it in the `io` package (I also think that `path.IntermediateAbsInfoField` should be there, in a file that contains useful definitions for Gobra that do not come from Isabelle). Additionally, I would call this function `UpdateCurrSeg` and I would make it a method for the type `IO_seg3`

Reply:
I move it to the `io` package and changed `UpdateCurrSeg.` I also renamed `IntermediateAbsInfoField` to `AbsInfoField`.